### ISSUE TITLE
swift-2.0 branch generating many errors (Alamofire issues)

### ISSUE THREAD

**Issue by zbeckman** [Moya/Moya#210]
Tried to pull the latest swift-2.0 into Xcode 7 GM and getting lots of compiler errors, such as "#" having been removed from Swift.

**Comment by ashfurrow** [comment]
That's odd. How are you installing it? You'll need to manually point to Alamofire's Swift 2 branch in your podfile. It's working for me with this one: https://github.com/artsy/eidolon/blob/master/Podfile

**Comment by zbeckman** [comment]
Tried a few different things, starting with the swift-2.0 branch README.md... obviously out of date.

After updating to:
```
pod 'Box', :head
pod 'Result', :head
pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :branch => 'swift-2.0'
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => 'https://github.com/Moya/Moya.git', :branch => 'swift-2.0'
```
It is healthier – seems to be down to two compile errors (versus the dozens prior to the change):
```
CompileSwift normal x86_64 /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/ReactiveCocoa/Moya+ReactiveCocoa.swift
    cd /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/Endpoint.swift -primary-file /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/ReactiveCocoa/Moya+ReactiveCocoa.swift /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/Moya.swift /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/ReactiveCore/MoyaResponse.swift /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/ReactiveCocoa/RACSignal+Moya.swift -target x86_64-apple-ios8.2 -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.0.sdk -I "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Products/QA 1-iphonesimulator" -F "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Products/QA 1-iphonesimulator" -g -import-underlying-module -module-cache-path /Users/zbeckman/Library/Developer/Xcode/DerivedData/ModuleCache -D COCOAPODS -serialize-debugging-options -Xcc "-I/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/swift-overrides.hmap" -Xcc -iquote -Xcc "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Moya-generated-files.hmap" -Xcc "-I/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Moya-own-target-headers.hmap" -Xcc "-I/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Moya-all-non-framework-target-headers.hmap" -Xcc -ivfsoverlay -Xcc /Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/all-product-headers.yaml -Xcc -iquote -Xcc "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Moya-project-headers.hmap" -Xcc "-I/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Products/QA 1-iphonesimulator/include" -Xcc -I/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Headers/Private -Xcc -I/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Headers/Private/Moya -Xcc -I/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Headers/Public -Xcc "-I/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/DerivedSources/x86_64" -Xcc "-I/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/DerivedSources" -Xcc -DQA_1=1 -Xcc -DCOCOAPODS=1 -Xcc -ivfsoverlay -Xcc "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/unextended-module-overlay.yaml" -Xcc -working-directory/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods -emit-module-doc-path "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Objects-normal/x86_64/Moya+ReactiveCocoa~partial.swiftdoc" -O -module-name Moya -emit-module-path "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Objects-normal/x86_64/Moya+ReactiveCocoa~partial.swiftmodule" -serialize-diagnostics-path "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Objects-normal/x86_64/Moya+ReactiveCocoa.dia" -emit-dependencies-path "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Objects-normal/x86_64/Moya+ReactiveCocoa.d" -emit-reference-dependencies-path "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Objects-normal/x86_64/Moya+ReactiveCocoa.swiftdeps" -o "/Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/Pods.build/QA 1-iphonesimulator/Moya.build/Objects-normal/x86_64/Moya+ReactiveCocoa.o"

/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/ReactiveCocoa/Moya+ReactiveCocoa.swift:37:66: error: value of type 'ErrorType' has no member 'domain'
                            subscriber.sendError(NSError(domain: error.domain, code: statusCode, userInfo: error.userInfo))
                                                                 ^~~~~ ~~~~~~
/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Moya/Moya/ReactiveCocoa/Moya+ReactiveCocoa.swift:39:50: error: cannot convert value of type 'ErrorType' to expected argument type 'NSError!'
                            subscriber.sendError(error)
                                                 ^~~~~
```
Ideas...?

By the way, what's the difference between your Alamofire branch and the Swift 2 master (Alamofire) branch? Can I safely point at the Alamofire URL or should I stick with your's for now...?

(Kind of looking for guidance on how to proceed now that Xcode 7 GM is out...)

**Comment by ashfurrow** [comment]
Hmm. I'll have to look at that tomorrow. Might be a change with the GM.

**Comment by jarrodrobins** [comment]
Alamofire were very quick - they've already merged their Swift branch into master and deleted the Swift 2 branch. You just need to point at the master branch for Alamofire now. 

I've submitted a PR for Moya which updates this.

**Comment by zbeckman** [comment]
Hmm. Still no good, I'm seeing the same error. Updated to use the latest AlamoFire. Here's the complete Podspec I'm using:
```
target 'Glimpulse', :exclusive => false do
        pod 'SwiftSpinner', :git => 'https://github.com/zbeckman/SwiftSpinner.git'
        pod 'XCGLogger', :git => 'https://github.com/DaveWoodCom/XCGLogger.git', :branch => 'swift_2.0'
        pod 'GRValidation', '~> 0.2'
        pod 'SwiftDate', :git => 'https://github.com/malcommac/SwiftDate.git', :branch => 'swift-2.0'
        pod 'DateTimeKit', :git => 'https://github.com/zbeckman/DateTimeKit.git'
        pod 'MagicalRecord'
        pod 'Box', :head
        pod 'Result', :head
        pod 'Alamofire', '~> 2.0'
        pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
        pod 'Moya/ReactiveCocoa', :git => 'https://github.com/Moya/Moya.git', :branch => 'swift-2.0'
end
```

**Comment by zbeckman** [comment]
(Realizing that, technically this looks like a Moya/ReactiveCocoa issue – I can create an issue over there if you would prefer).

**Comment by ashfurrow** [comment]
I'm going to work on getting it all to work this afternoon with the Xcode 7 GM.

**Comment by ashfurrow** [comment]
OK cool, so this is working for me ([this PR](https://github.com/Moya/Moya/pull/212)).

@zbeckman Your podfile looks good to me. But you'll need to update things:

```sh
pod update Box Result Alamofire Moya/ReactiveCocoa
```

Let us know if that works :cake:

**Comment by ashfurrow** [comment]
The code on this PR is working for me locally: https://github.com/artsy/eidolon/pull/504

**Comment by zbeckman** [comment]
I'm now pulling the following pods:
```
	pod 'Box', :head
	pod 'Result', :head
	pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
	pod 'Alamofire', '~> 2.0'
	pod 'Moya', :git => 'https://github.com/Moya/Moya.git', :branch => 'swift-2.0'
```
After adding Alamofire and Moya, I'm getting the following runtime error. This only *seems* to be happening after adding in Alamofire and Moya, but I'm still debugging.

If you see anything wrong with my config please let me know.
```
2015-09-11 01:48:07.648 XCTRunner[35507:240964] The bundle “GlimpulseUITests” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
2015-09-11 01:48:07.648 XCTRunner[35507:240964] (dlopen_preflight(/Users/zbeckman/Library/Developer/CoreSimulator/Devices/FCFAAFBF-1AB4-4223-9BCC-CA91CBDE96DC/data/Containers/Bundle/Application/11F5D849-3061-4E91-A44C-F35AA544520A/GlimpulseUITests-Runner.app/PlugIns/GlimpulseUITests.xctest/GlimpulseUITests): Library not loaded: @rpath/libswiftCoreLocation.dylib
  Referenced from: /Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/CodeCoverage/Glimpulse/Products/QA 1-iphonesimulator/ReactiveCocoa.framework/ReactiveCocoa
  Reason: image not found)
```

**Comment by ashfurrow** [comment]
Oh no! I had a similar issue yesterday but I can't recall exactly what fixed it. You've tried the routine fixes like deleting derived data and resetting the simulator? Are there any compiler warnings that seem related?

**Comment by zbeckman** [comment]
Yes – all the routine fixes.

I'm very fuzzy on which pods I need to pull in... ReactiveCocoa, Alamofire, Moya, ReactiveMoya... And I'm pretty sure the README is quite out of date in this regard.

I'm able to get Moya to install – but, any attempt to bring in ReactiveCocoa fails.

**Comment by orta** [comment]
Here is our working build of [Moya + ReactiveCocoa](https://github.com/artsy/eidolon/blob/master/Podfile#L65-L67) - which generates this [lockfile](https://github.com/artsy/eidolon/blob/master/Podfile.lock)

``` ruby
pod 'Alamofire'
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => 'https://github.com/Moya/Moya.git', :branch => 'swift-2.0'
```

Should get you the right setup, if you have been using older references for these branches - you may need to `pod install`

**Comment by ashfurrow** [comment]
I agree the README needs updating – I'll roll that into https://github.com/Moya/Moya/pull/214

**Comment by zbeckman** [comment]
Hm, OK – a little more info – the error I reported above:
```
2015-09-11 01:48:07.648 XCTRunner[35507:240964] The bundle “GlimpulseUITests” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
2015-09-11 01:48:07.648 XCTRunner[35507:240964] (dlopen_preflight(/Users/zbeckman/Library/Developer/CoreSimulator/Devices/FCFAAFBF-1AB4-4223-9BCC-CA91CBDE96DC/data/Containers/Bundle/Application/11F5D849-3061-4E91-A44C-F35AA544520A/GlimpulseUITests-Runner.app/PlugIns/GlimpulseUITests.xctest/GlimpulseUITests): Library not loaded: @rpath/libswiftCoreLocation.dylib
  Referenced from: /Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/CodeCoverage/Glimpulse/Products/QA 1-iphonesimulator/ReactiveCocoa.framework/ReactiveCocoa
  Reason: image not found)
```
Only seems to be a problem in the "iOS UI Testing Bundle" target type. (Unfortunately, this is where we do all our testing, so the outcome is that when we run tests, it immediately fails and nothing works – we just get the above error).

**Comment by zbeckman** [comment]
Also, if I only use:
```
pod 'Alamofire'
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => 'https://github.com/Moya/Moya.git', :branch => 'swift-2.0'
```
Then, I get compile errors as follows:
```
CompileSwift normal x86_64 /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Box/Box/Box.swift
    cd /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods

/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Box/Box/Box.swift:6:37: error: 'Printable' has been renamed to 'CustomStringConvertible'
public final class Box<T>: BoxType, Printable {
                                    ^~~~~~~~~
                                    CustomStringConvertible
/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Box/Box/Box.swift:31:10: error: 'toString' has been renamed to 'String'
                return toString(value)
                       ^~~~~~~~
                       String
Swift.toString:1:33: note: 'toString' has been explicitly marked unavailable here
@available(*, unavailable) func toString<T>(x: T) -> String
                                ^

CompileSwift normal x86_64 /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Box/Box/MutableBox.swift
    cd /Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods
/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Box/Box/MutableBox.swift:8:51: error: 'Printable' has been renamed to 'CustomStringConvertible'
public final class MutableBox<T>: MutableBoxType, Printable {
                                                  ^~~~~~~~~
                                                  CustomStringConvertible
/Users/zbeckman/Projects/Glimpulse/iOS/project/Glimpulse/Pods/Box/Box/MutableBox.swift:25:10: error: 'toString' has been renamed to 'String'
                return toString(value)
                       ^~~~~~~~
                       String
Swift.toString:1:33: note: 'toString' has been explicitly marked unavailable here
@available(*, unavailable) func toString<T>(x: T) -> String
                                ^
```
(Hence, the addition of the pods for Box and Result from HEAD). However, once I add in those (fixing the compiler errors) – we end up back at the original problem where none of our tests work (the `image not found` error).

**Comment by zbeckman** [comment]
It would seem that if I build ReactiveCocoa using Carthage, and install it as directed (that is, drag the framework into my embedded binaries) – the problem changes. Now:
```
2015-09-11 21:15:50.796 XCTRunner[20612:2543548] (dlopen_preflight(/Users/zbeckman/Library/Developer/CoreSimulator/Devices/FCFAAFBF-1AB4-4223-9BCC-CA91CBDE96DC/data/Containers/Bundle/Application/7FD22288-32A0-419F-B0AF-54DE9143FDBB/GlimpulseUITests-Runner.app/PlugIns/GlimpulseUITests.xctest/GlimpulseUITests): Library not loaded: @rpath/libswiftCoreLocation.dylib
  Referenced from: /Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/CodeCoverage/Glimpulse/Products/QA 1-iphonesimulator/Moya.framework/Moya
  Reason: image not found)
```
(Notice that now it's Moya that has library problems...)

I think this has something to do with the fact that I'm building &/or testing for iOS. And I found this, which I'm not sure what to make of: https://github.com/Carthage/Carthage#if-youre-building-for-ios.

**Comment by ashfurrow** [comment]
I've been able to use CocoaPods to install Moya/ReactiveMoya without any problems on a fresh Xcode project (download it [here](http://cl.ly/433l3C18233J)). I did so using Xcode 7 GM and the following Podfile:

```rb
use_frameworks!

target 'Test' do

pod 'Result', :head
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => 'https://github.com/Moya/Moya.git'

end
```

@zbeckman If that isn't working for you on your existing project, could you try cleaning your build directory? Command-Shift-K should do it in Xcode. It could be that Xcode's forgotten its derived data cache is stale. Let us know if any of that helps.

**Comment by zbeckman** [comment]
Ok, so after a good bit of fiddling, I learned a few things.

I'm using Xcode 7 GM. I'm in the habit of clearing the project directory and derived data whenever anything odd happens (so, um... like hourly ;-).

First off, everything works just fine if I install ReactiveCocoa using Carthage (Swift 2 branch of course), and drag the framework into our project, in conjunction with the following pods (all required for a good build). In addition, I have `SwiftCheck` installed via Carthage:
```
# UX components
pod 'SwiftSpinner', :git => 'https://github.com/zbeckman/SwiftSpinner.git'

# Frameworks
pod 'MagicalRecord'
pod 'Alamofire'
pod 'Moya', :git => 'https://github.com/Moya/Moya.git', :branch => 'swift-2.0'

# Utility components
pod 'XCGLogger', :git => 'https://github.com/DaveWoodCom/XCGLogger.git', :branch => 'swift_2.0'
pod 'GRValidation', '~> 0.2'
pod 'DateTimeKit', :git => 'https://github.com/zbeckman/DateTimeKit.git'
```
I also learned that `BrightFutures` has an incompatible dependency (it requires `Result` 0.6.1 and ReactiveCocoa is now using 1.0.0 – which leads to a lot of problems... However, with RAC your probably don't need BrightFutures so hopefully this isn't an issue for anyone).

*NOW.* After getting all that figured out (and committed! It took forever so I don't want to lose this config), I gave our project a try with *nothing except these pods* (and, of course, I removed the Carthage install of RAC):
```
pod 'SwiftSpinner', :git => 'https://github.com/zbeckman/SwiftSpinner.git'
pod 'Result', :head
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => 'https://github.com/Moya/Moya.git'
pod 'XCGLogger', :git => 'https://github.com/DaveWoodCom/XCGLogger.git', :branch => 'swift_2.0'
pod 'GRValidation', '~> 0.2'

target 'GlimpulseTests', :exclusive => false do
	# Testing framework
	pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'swift-2.0'
	pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
end
```
Did the usual (clean project, delete derived data) – **and it fails** with the following output:
```
2015-09-12 20:20:32.726 XCTRunner[84436:3618319] (dlopen_preflight(/Users/zbeckman/Library/Developer/CoreSimulator/Devices/FCFAAFBF-1AB4-4223-9BCC-CA91CBDE96DC/data/Containers/Bundle/Application/EF1E312C-E955-45EE-B805-B86307D314A2/GlimpulseUITests-Runner.app/PlugIns/GlimpulseUITests.xctest/GlimpulseUITests): Library not loaded: @rpath/libswiftCoreLocation.dylib
  Referenced from: /Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/CodeCoverage/Glimpulse/Products/QA 1-iphonesimulator/Moya.framework/Moya
  Reason: image not found)
```
Just for the heck of it (to make sure nothing is different) I tried:
```
pod 'Moya', :git => 'https://github.com/Moya/Moya.git'
```
But I think that initial name is just symbolic – anyhow, it didn't matter.

So bottom line – Still no good using the CocoaPod, but it's fine using Carthage from the original ReactiveCocoa repository.

I'm stumped...

**Comment by ashfurrow** [comment]
I've never used Carthage and CocoaPods together before – not sure of how that'd work. What version of CocoaPods are you using? There's a similar issue in #216.

**Comment by ashfurrow** [comment]
We've rearranged this repository since you initially had this problem. Could I ask you to try creating a new Xcode project, and `pod install` using the following pods? Please leave Carthage out so that we can narrow the problem down. 

```rb
pod 'Result', :head
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => "https://github.com/Moya/Moya.git"
```

After `pod install`ing, open the project in Xcode 7 GM and run the app. Hopefully it'll work (it's working for me). 

If that doesn't work, please download [this Xcode project](http://cl.ly/3e1f342x2G2X) which I've prepared. It works on my machine. If it doesn't on your machine, we'll have a better idea of where to look. 

If that _does_ work, we'll try the same approach on your library. If you could undo the changes made when you were troubleshooting using git, that would be helpful. If it still doesn't work, then it could be something specific about your project. 

I'll continue looking for the solution on my end. Thanks for helping us figure out the problem! :cake:

**Comment by zbeckman** [comment]
I tried removing the Carthage install of ReactiveCocoa, and then updating with the pods you suggested:
```
pod 'Result', :head
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => "https://github.com/Moya/Moya.git"
```
I get the same problem, again, only on the UI testing target (regular test targets and the app itself seem to work fine):
```
2015-09-13 11:22:13.386 XCTRunner[29003:4857736] (dlopen_preflight(/Users/zbeckman/Library/Developer/CoreSimulator/Devices/FCFAAFBF-1AB4-4223-9BCC-CA91CBDE96DC/data/Containers/Bundle/Application/056B5BED-0B9F-41AB-A248-117BD6824748/GlimpulseUITests-Runner.app/PlugIns/GlimpulseUITests.xctest/GlimpulseUITests): Library not loaded: @rpath/libswiftCoreLocation.dylib
  Referenced from: /Users/zbeckman/Library/Developer/Xcode/DerivedData/Glimpulse-avhroqmcgyzbiecltqxshdpczkht/Build/Intermediates/CodeCoverage/Glimpulse/Products/QA 1-iphonesimulator/Moya.framework/Moya
  Reason: image not found)
```
I also took a look at your test project. There is an important different in regard to how we have our Podfiles set up, and possibly the resulting code:

My Podfile installs all of the common pods without including them in a target (that is, like this):
```
pod 'Moya'
pod 'OtherCommonStuff'

target 'UITests'
    pod 'Quick'
end
```
So, with that in mind, I started "upgrading" your Podfile, incrementally adding elements until I got a similar failure.

The first step was to adopt the above format. Then I added a few lines that were missing from yours, but present in my config (such as the iOS target). Finally, I added in my testing libraries.

This is what I ended up with:
```
platform :ios, '8.0'

source 'https://github.com/CocoaPods/Specs.git'

use_frameworks!

pod 'Result', :head
pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
pod 'Moya/ReactiveCocoa', :git => "https://github.com/Moya/Moya.git"

target 'TestTests' do
	pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'swift-2.0'
	pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
end

target 'TestUITests' do
	pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'swift-2.0'
	pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
end
```
And I get this result when the UI tests run:
```
2015-09-13 11:31:46.948 XCTRunner[32697:4885511] (dlopen_preflight(/Users/zbeckman/Library/Developer/CoreSimulator/Devices/A5E91AA9-62B2-43C9-9378-9FA9A3F5491D/data/Containers/Bundle/Application/B2691E9B-944A-49C7-8EF3-FE1CCB98D62B/TestUITests-Runner.app/PlugIns/TestUITests.xctest/TestUITests): Library not loaded: @rpath/libswiftCoreLocation.dylib
  Referenced from: /Users/zbeckman/Library/Developer/Xcode/DerivedData/Test-hbfuhujgtjfczffhoidfubqaazyo/Build/Products/Debug-iphonesimulator/Moya.framework/Moya
  Reason: image not found)
```

**Comment by ashfurrow** [comment]
Great work! I'm having trouble with the Demo project in #216 too. After adding pods to the specific target in the Podfile, it's now working! This kind of stuff has always tripped me up – looping in @orta.

**Comment by zbeckman** [comment]
Same here! Ok, I updated our main project's Podfile as follows (specifically, I put the main target 'Glimpulse' back in, so now every pod is explicitly included in a target):
```
platform :ios, '8.0'
source 'https://github.com/CocoaPods/Specs.git'
use_frameworks!

target 'Glimpulse', :exclusive => false do
	# UX components
	pod 'SwiftSpinner', :git => 'https://github.com/zbeckman/SwiftSpinner.git'

	# Frameworks
	pod 'MagicalRecord'
	pod 'Result', :head
	pod 'ReactiveCocoa', :git => 'https://github.com/ashfurrow/ReactiveCocoa.git', :branch => 'swift2'
	pod 'Moya/ReactiveCocoa', :git => "https://github.com/Moya/Moya.git"

	# Utility components
	pod 'XCGLogger', :git => 'https://github.com/DaveWoodCom/XCGLogger.git', :branch => 'swift_2.0'
	pod 'GRValidation', '~> 0.2'
	pod 'DateTimeKit', :git => 'https://github.com/zbeckman/DateTimeKit.git'
end

target 'GlimpulseTests', :exclusive => false do
	# Testing framework
	pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'swift-2.0'
	pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
end

target 'GlimpulseIntegrationTests', :exclusive => false do
	# Testing framework
	pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'swift-2.0'
	pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
end

target 'GlimpulseUITests', :exclusive => false do
	# Testing framework
	pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'swift-2.0'
	pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
end
```
And it works!

Questions, though:

Notice that *I am not including* any of the following pods:
```
	#pod 'Box', :head
	#pod 'Result', :head
	#pod 'Alamofire'
```
So, are `Result` and `Alamofire` looped in through dependencies? Do we need to include these in our Podfile?

Thankfully this simplifies our Carthage setup too... just one project now (`SwiftCheck`) that doesn't do pods.

**Comment by ashfurrow** [comment]
Great! I'm glad that's working. 

Box is no longer required at all. Result seems to have been updated on Trunk, so we can remove it too. Same with Alamofire – used to need it specified for Swift 2 support, don't anymore.

**Comment by orta** [comment]
BTW you can reduce test duplication by doing [something like this](https://github.com/orta/Elasticity/blob/master/Podfile#L31-L38).

**Comment by zbeckman** [comment]
Nice! Thanks for the tip.

**Comment by ashfurrow** [comment]
Seems like this is resolved, I'm going to close the issue. Don't hesitate to reopen, or open a new issue, if you run into any further problems. Thanks again for your help! :100:
